### ISSUE TITLE
performance: Level.getWeight

### ISSUE THREAD

**Issue by tyrcho** [skjolber/3d-bin-container-packing#486]
I have noticed on our production server which makes heavy use of this library (version 1.2.14 for the moment) that the top method by CPU usage is `Level.getWeight`

Could this method be improved ? I think we should probably store the weight (and height) in the Level, and only recompute it when an element is added to the ArrayList.

![image](https://user-images.githubusercontent.com/700260/162405491-e4c13bd7-203d-4cce-b9e7-074080d3329d.png)

**Comment by skjolber** [comment]
Please create a PR again [1.x](https://github.com/skjolber/3d-bin-container-packing/tree/1.x).

**Comment by tyrcho** [comment]
#504 

I finally found some time to try an implementation for this :) 

I've got some random failures in the PBT, but my guess is they are more related to some variant of #503 than this change.

```
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 3.32 s <<< FAILURE! - in BruteForcePropertyBasedTests
[ERROR] identicalBoxesShouldFitInContainers  Time elapsed: 2.916 s  <<< ERROR!
java.lang.IllegalArgumentException: Remaining free space is negative at -2 for Container [stackWeight=3, stackHeight=5, levels=[com.github.skjolber.packing.Level@6f03bf66], weight=8, width=2, depth=4, height=6, volume=48, name=null]
        at com.github.skjolber.packing.BruteForcePropertyBasedTests.identicalBoxesShouldFitInContainers(BruteForcePropertyBasedTests.java:44)

```

**Comment by skjolber** [comment]
Those property based unit tests are doing their job :+1: :+1:

**Comment by skjolber** [comment]
Thank for the PR, released as 1.2.16
